sormas-ui/src/main/java/de/symeda/sormas/ui/person/PersonEditForm.java

```diff
@@ -678,38 +691,54 @@ private void onEmancipatedChange() {
 		if(isEmancipatedChecked) {
 			nameOfGuardians.setValue("");
 		}
-		updateHasGuardianCheckBox();
+		updateHasGuardianCheckBox(isEmancipatedChecked);
 	}
 
 	private void updateIsIncapacitatedCheckBox(boolean isInitialized) {
```

Comment:
this method can become the main method we were talking we should group the legal guardian logic into. Since it does a lot more that its name states please give it a more suggestive name. Something like "updateLegalGuardianSection"